test/e2e/specs/specs-playwright/wp-signup__free.ts

```diff
@@ -114,15 +116,30 @@ describe(
 		} );
 
 		describe( 'Delete user account', function () {
-			it( 'Re-login to ensure correct host', async function () {
-				await BrowserManager.clearAuthenticationState( page );
-				const loginPage = new LoginPage( page );
-				await loginPage.visit();
-				await loginPage.login( { username: username, password: signupPassword } );
+			let newPage: Page;
+
+			afterAll( async function () {
+				await BrowserManager.closePage( newPage, { closeContext: true } );
+			} );
+
+			it( 'Launch new context to ensure correct host', async function () {
+				newPage = await BrowserManager.newPage( { newContext: true } );
+			} );
+
+			it( 'Login', async function () {
+				// Logging in and immediately using navbar often leads to race conditions when the redirect
+				// from <rooturl> to <rooturl>/home/<site> redirect takes you out of the navigation you just did.
+				// We need to wait for that to happen before proceeding.
+				const expectedLandngUrl = DataHelper.getCalypsoURL( `home/${ selectedDomain }` );
```

Comment:
I see why it has to be done, but something like this work?

```
await Promise.all([
    page.waitforNavigation( expectedLandingURL),
    loginFlow.logIn()
])
```

Reply:
Oh, that does seem a lot nicer! I will move that out of the `login` function.

Do we want to remove the internal `waitForNavigation` in the `login` function?

Reply:
The internal `waitForNavigation` needs to be kept, otherwise the method will not wait until the cookie is set + home dashboard loads before executing the next step, I think.

Reply:
Even if you have two layers of `waitForNavigation`, it _should not_ interfere with one another...but I haven't verified this.

Reply:
Fixed in [c77987d](https://github.com/Automattic/wp-calypso/pull/56469/commits/c77987dd0f4e5a42c0bed807471796c59bafca17)

Reply:
>Even if you have two layers of waitForNavigation, it should not interfere with one another...but I haven't verified this.

Yep, you're totally right! Good catch!